case_studies/redshift/network_rs.py

```diff
@@ -0,0 +1,99 @@
+# Author: Qiaozhi Huang
+# Class for network to predict redshift prediction
+import numpy as np
+import pytorch_lightning as pl
+import torch
+import torch.nn as nn
+from tqdm import tqdm
+
+
+class Regressor(nn.Module):
+    def __init__(self,
+        in_dim,                             # Number of input features.
+        hidden_dim,                         # Number of input features.
+        out_dim,                            # Number of output features.
+        dropout_rate=0.5,
+    ):
+        super().__init__()
+        self.in_dim = in_dim
+        self.hidden_dim = hidden_dim
+        self.out_dim = out_dim
+
+        self.net = nn.Sequential(
+            nn.Linear(in_dim, hidden_dim),
+            # nn.BatchNorm1d(hidden_dim),
+            # nn.LayerNorm(hidden_dim),
+            nn.ReLU(),
+            # nn.Dropout(dropout_rate),
+            nn.Linear(hidden_dim, hidden_dim),
+            # nn.BatchNorm1d(hidden_dim),
+            # nn.LayerNorm(hidden_dim),
+            nn.ReLU(),
+            nn.Dropout(dropout_rate),
+            nn.Linear(hidden_dim, hidden_dim),
+            # nn.BatchNorm1d(hidden_dim),
+            # nn.LayerNorm(hidden_dim),
+            nn.ReLU(),
+            # nn.Dropout(dropout_rate),
+            nn.Linear(hidden_dim, hidden_dim),
+            # nn.BatchNorm1d(hidden_dim),
+            # nn.LayerNorm(hidden_dim),
+            nn.ReLU(),
+            # nn.Dropout(dropout_rate),
+            nn.Linear(hidden_dim, out_dim),
+        )
+
+    def forward(self, x):
+        return self.net(x)
+
+
+class LitRegressor(pl.LightningModule):
```

Comment:
It'd be better to merge `Regression` and `LitRegression` into a single subclass of `pl.LightningModule`